File: core/test/geo_transforms.cpp
#include <iomanip>
#include <iostream>
#include <random>
#include <string>
#include <cstdlib>

#include <GeographicLib/LocalCartesian.hpp>
#include <CoverageControl/geographiclib_wrapper.h>

int main(int argc, char** argv) {
	CoverageControl::GeoLocalTransform geo_transform(40.74050005471615, -74.1759877275644, 0);
	auto lla = geo_transform.Reverse(100, 100, 0);
	std::cout << "lat: " << lla[0] << std::endl;
	std::cout << "lon: " << lla[1] << std::endl;
	std::cout << "hgt: " << lla[2] << std::endl;

	auto xyz = geo_transform.Forward(lla[0], lla[1], lla[2]);
	std::cout << "x: " << xyz[0] << std::endl;
	std::cout << "y: " << xyz[1] << std::endl;
	std::cout << "z: " << xyz[2] << std::endl;

	GeographicLib::LocalCartesian geo_transform1(40.74050005471615, -74.1759877275644);
	geo_transform1.Forward(lla[0], lla[1], lla[2], xyz[0], xyz[1], xyz[2]);
	std::cout << "x: " << xyz[0] << std::endl;
	std::cout << "y: " << xyz[1] << std::endl;
	std::cout << "z: " << xyz[2] << std::endl;

	return 0;
}

File: python/tests/geo_transforms.py
import numpy as np
import pyCoverageControl
from pyCoverageControl import GeoLocalTransform

# Provide lat, lon, altitude as the origin
geo_transform = GeoLocalTransform(40.74050005471615, -74.1759877275644, 0)

# Reverse() takes in x, y, z relative to the origin and gives out [lat, lon, altitude]
lla = geo_transform.Reverse(100, 100, 0);
print(lla)

# Forward() takes in lat, lon, altitude and gives out [x, y, z] relative to origin
xyz = geo_transform.Forward(lla[0], lla[1], lla[2]);
print(xyz) # Should be [100 100 0]
